pkgs/test_core/lib/src/direct_run.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:path/path.dart' as p;
+import 'package:test_api/backend.dart'; //ignore: deprecated_member_use
+import 'package:test_api/src/backend/declarer.dart'; //ignore: implementation_imports
+import 'package:test_api/src/backend/group.dart'; //ignore: implementation_imports
+import 'package:test_api/src/backend/group_entry.dart'; //ignore: implementation_imports
+import 'package:test_api/src/backend/invoker.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/test.dart'; //ignore: implementation_imports
+import 'package:test_api/src/utils.dart'; // ignore: implementation_imports
+
+import 'runner/configuration.dart';
+import 'runner/engine.dart';
+import 'runner/plugin/environment.dart';
+import 'runner/reporter.dart';
+import 'runner/reporter/expanded.dart';
+import 'runner/runner_suite.dart';
+import 'runner/suite.dart';
+import 'util/print_sink.dart';
+
+/// Run all test cases declared in [testMain].
+///
+/// Test suite level metadata defined in annotations is not read. No filtering
+/// is applied except for the filtering defined by `solo` or `skip` arguments to
+/// `group` and `test`.
+Future<bool> directRunTests(FutureOr<void> Function() testMain,
+        {Reporter Function(Engine)? reporterFactory}) =>
+    _directRunTests(testMain, reporterFactory: reporterFactory);
+
+/// Run a single test declared in [testMain].
+///
+/// There must be exactly one test defined with the name [testName]. Note that
+/// not all tests and groups are checked, so a test case that may not be
+/// intended to be run (due to a `solo` on a different test) may still be run
+/// with this API. Only the test names returned by [enumerateTestCases] should
+/// be used.
+Future<bool> directRunSingleTest(
+        FutureOr<void> Function() testMain, String testName,
+        {Reporter Function(Engine)? reporterFactory}) =>
+    _directRunTests(testMain,
+        reporterFactory: reporterFactory, runSingleTestByName: testName);
+
+Future<bool> _directRunTests(FutureOr<void> Function() testMain,
+    {Reporter Function(Engine)? reporterFactory,
+    String? runSingleTestByName}) async {
```

Comment:
I would also line up the name here with the field name on Declarer

Reply:
Done